### PR TITLE
Read the entire artifacts from GCS

### PR DESCRIPTION
**Pull request by Katharine**
Some Spyglass codepaths featured an erroneous assumption that `Read()` would always return the expected number of bytes, assuming at least that many bytes exist. This is not how reading files works, and so we were ending up with buffers full of trailing zeros, which manifested as a lot of question marks, sometimes.

Add a loop to keep reading until we run out of space, and a unit test that forces us to actually do this.

/cc @stevekuznetsov

### CODE REVIEW

File: prow/spyglass/gcsartifact.go [hunk 1]
```diff
@@ -117,14 +117,23 @@ func (a *GCSArtifact) ReadAt(p []byte, off int64) (n int, err error) {
...
 		return 0, fmt.Errorf("error getting artifact reader: %v", err)
 	}
-	n, err = reader.Read(p)
-	if err != nil {
-		return 0, fmt.Errorf("error reading from artifact: %v", err)
+	// We need to keep reading until we fill the buffer or hit EOF.
+	offset := 0
+	for offset < len(p) {
+		n, err = reader.Read(p[offset:])
+		fmt.Println("Read ", n, p)
```
Comment (stevekuznetsov) [inline prow/spyglass/gcsartifact.go]: Is this a leftover debug?

Comment (Katharine) [reply prow/spyglass/gcsartifact.go]: Yup! Whoops

Comment (Katharine) [reply prow/spyglass/gcsartifact.go]: Fixed.

**Pull request merged**